lib/Optimizer/CodeGen/LowerToQIR.cpp

```diff
@@ -1652,6 +1652,10 @@ void cudaq::opt::initializeTypeConversions(LLVMTypeConverter &typeConverter) {
     auto eleTy = typeConverter.convertType(type.getElementType());
     return factory::stdVectorImplType(eleTy);
   });
+  typeConverter.addConversion([&typeConverter](cc::CharspanType type) {
+    auto eleTy = typeConverter.convertType(type.getElementType());
+    return factory::stdStringImplType(eleTy);
```

Comment:
Why use `stdStringImplType` or why does `stdStringImplType` return a `{elePtrTy, i64Ty, i64Ty, i64Ty}`? Shouldn't a span type just be `{elePtrTy, i64Ty}`?

Reply:
My bad. We should be using the span form here. Doh!